LoanCalc.java

```diff
@@ -28,27 +28,60 @@ public static void main(String[] args) {
 	// Computes the ending balance of a loan, given the loan amount, the periodical
 	// interest rate (as a percentage), the number of periods (n), and the periodical payment.
 	private static double endBalance(double loan, double rate, int n, double payment) {	
-		// Replace the following statement with your code
-		return 0;
+		double loan1 = loan;
+		double rate1 =  rate / 100;
+		int n1 = n;
+		for( int i = 0 ; i < n1 ; i++){
+			loan1 = loan1 - payment;
+			loan1 = ( loan1 ) * (1 + rate1);
+		}
+		return loan1;
 	}
+
 	
 	// Uses sequential search to compute an approximation of the periodical payment
 	// that will bring the ending balance of a loan close to 0.
 	// Given: the sum of the loan, the periodical interest rate (as a percentage),
 	// the number of periods (n), and epsilon, the approximation's accuracy
 	// Side effect: modifies the class variable iterationCounter.
     public static double bruteForceSolver(double loan, double rate, int n, double epsilon) {
-		// Replace the following statement with your code
-		return 0;
+		double loan1 = loan;
```

Comment:
You need to reset iteration counter to 0... 
Assume you run bruteForceSolver twice...